src/main/java/me/botsko/prism/actions/BlockAction.java

```diff
@@ -350,50 +350,52 @@ ChangeResult placeBlock(Player player, QueryParameters parameters, boolean isPre
+                ((CreatureSpawner) newState).setDelay(s.getDelay());
+                ((CreatureSpawner) newState).setSpawnedType(s.getEntityType());
 
-        }
+            }
 
-        if (getMaterial() == COMMAND_BLOCK
-                && blockActionData instanceof CommandActionData) {
-            final CommandActionData c = (CommandActionData) blockActionData;
-            ((CommandBlock) newState).setCommand(c.command);
-        }
-        if (newState instanceof Nameable && actionData.customName != null) {
-            ((Nameable) newState).setCustomName(actionData.customName);
-        }
-        if (parameters.getProcessType() == PrismProcessType.ROLLBACK
-                && Tag.SIGNS.isTagged(getMaterial())
-                && blockActionData instanceof SignActionData) {
-
-            final SignActionData s = (SignActionData) blockActionData;
-            // Verify block is sign. Rarely, if the block somehow pops off
-            // or fails
-            // to set it causes ClassCastException:
-            // org.bukkit.craftbukkit.v1_4_R1.block.CraftBlockState
-            // cannot be cast to org.bukkit.block.Sign
-            // https://snowy-evening.com/botsko/prism/455/
-            if (newState instanceof Sign) {
-                if (s.lines != null) {
-                    for (int i = 0; i < s.lines.length; ++i) {
-                        ((Sign) newState).setLine(i, s.lines[i]);
+            if (getMaterial() == COMMAND_BLOCK
+                    && blockActionData instanceof CommandActionData) {
+                final CommandActionData c = (CommandActionData) blockActionData;
+                ((CommandBlock) newState).setCommand(c.command);
+            }
+            if (newState instanceof Nameable && blockActionData.customName != null && !blockActionData.customName.equals("")) {
+                ((Nameable) newState).setCustomName(blockActionData.customName);
+            }
+            if (parameters.getProcessType() == PrismProcessType.ROLLBACK
+                    && Tag.SIGNS.isTagged(getMaterial())
+                    && blockActionData instanceof SignActionData) {
+
+                final SignActionData s = (SignActionData) blockActionData;
+                // Verify block is sign. Rarely, if the block somehow pops off
+                // or fails
+                // to set it causes ClassCastException:
+                // org.bukkit.craftbukkit.v1_4_R1.block.CraftBlockState
+                // cannot be cast to org.bukkit.block.Sign
+                // https://snowy-evening.com/botsko/prism/455/
+                if (newState instanceof Sign) {
+                    if (s.lines != null) {
+                        for (int i = 0; i < s.lines.length; ++i) {
+                            ((Sign) newState).setLine(i, s.lines[i]);
+                        }
                     }
                 }
             }
+        } else {
+            Prism.debug("BlockAction Data was null with "+parameters.toString());
```

Comment:
Codacy found an issue: [WhitespaceAround: '+' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5911179)